### DISCUSSION TITLE
Incorrect duration/length for streaming .m3u8

### DISCUSSION THREAD

**Discussion by wankio** [streamlink/streamlink#5263] [Q&A] [answered]
### Description

Today when i'm use streamlink again to download some m3u8 files, it gave me wrong duration/length
Mediainfo is correct, no problems

incorrect length/duration only happens if u playing it with any media player
eg, mpv logs : 
```
 (+) Video --vid=1 (*) (h264 1280x720 30.000fps)
 (+) Audio --aid=1 (*) (aac 2ch 48000Hz)
AO: [wasapi] 48000Hz stereo 2ch float
VO: [gpu] 1280x720 yuv420p
AV: 00:09:44 / 00:11:17 (86%) A-V:  0.000
Invalid audio PTS: 578.880000 -> 584.085333
Invalid audio PTS: 584.128000 -> 585.429333
Invalid audio PTS: 585.472000 -> 586.773333
Invalid audio PTS: 586.794667 -> 588.096000
Invalid audio PTS: 588.138667 -> 589.440000
Invalid audio PTS: 589.482667 -> 590.784000
Reset playback due to audio timestamp reset.
[ffmpeg/video] h264: reference picture missing during reorder
[ffmpeg/video] h264: Missing reference picture, default is 0
[ffmpeg/video] h264: mmco: unref short failure
^C: 00:08:03 / 00:11:17 (71%) A-V:  0.000
```

I fixed it by use ffmpeg -c:v copy -c:a copy, it gave me correct length when playing with any media player

correct length : Duration : 3 min 16 s
length playing with any media player: 11 min 17s

If i'm remember this causes by desync audio in streaming file on server-side right ? Thank

This is first time i have this problem

**Comment by bastimeyer** [comment, marked as answer]
Post the link of the HLS stream you've downloaded. What you've posted is way too vague and include any actual information.

Streamlink's output of HLS streams is a simple concatenation of MPEG-TS or fragmented MP4 segments. Streamlink does no remuxing unless an external audio or subtitle stream needs to be included in the output. This means that if there are presentation timestamp issues in the audio stream, this is most likely the fault of the stream itself.

**Reply by wankio** [reply]
there : nsfw REDACTED

**Reply by bastimeyer** [reply]
Regular segment concatenation with fragmented MPEG-4 containers. No muxing is being done, so if there are DTS/PTS issues, then it's because of the stream itself.

Also locking the thread, since we don't support NSFW content, as explicitly stated in the contributing guidelines.
